modules/imgcodecs/src/loadsave.cpp

```diff
@@ -721,9 +721,9 @@ static bool imwrite_( const String& filename, const std::vector<Mat>& img_vec,
         CV_Assert( image.channels() == 1 || image.channels() == 3 || image.channels() == 4 );
 
         Mat temp;
-        if( !encoder->isFormatSupported(image.depth()) )
+        if( (!encoder->isFormatSupported(image.depth())) && (encoder->isFormatSupported(CV_8U)) )
         {
-            CV_Assert( encoder->isFormatSupported(CV_8U) );
+            CV_LOG_ONCE_WARNING(NULL, "Unsupported depth image for selected encoder is fallbacked to CV_8U.");
             image.convertTo( temp, CV_8U );
```

Comment:
Log is good, but we get U.B., if the encoder does not support CV_8U by some reason.

Reply:
Thank you for your review. This change comes from https://github.com/opencv/opencv/pull/26211#discussion_r1779455750 .

If there are no problem, I agree with revert it.

```cpp
        if( !encoder->isFormatSupported(image.depth() ) )
        {
            CV_LOG_ONCE_WARNING(NULL, "Unsupported depth image for selected encoder is fallbacked to CV_8U.");
            CV_Assert( encoder->isFormatSupported(CV_8U) );
            image.convertTo( temp, CV_8U );
```

Reply:
Thank you for your review, I fixed it.